tracer/test/Datadog.Trace.Tests/Tagging/TagsListTests.cs

```diff
@@ -86,38 +74,43 @@ public void CheckProperties()
         [Fact]
         public void Serialization_RootSpan()
         {
-            var tracer = new Mock<IDatadogTracer>();
-            var traceContext = new TraceContext(tracer.Object);
+            var tracer = new MockTracer();
+            var traceContext = new TraceContext(tracer);
             var span = new Span(new SpanContext(SpanContext.None, traceContext, "service1"), start: null);
             traceContext.AddSpan(span);
 
             const int customTagCount = 15;
             SetupForSerializationTest(span, customTagCount);
-            var deserializedSpan = SerializeSpan(span);
+            span.Finish();
+
+            var deserializedSpan = Serialize(tracer.TraceChunk, traceContext).Single();
 
-            deserializedSpan.Tags.Count.Should().Be(customTagCount + 4);
             deserializedSpan.Tags.Should().Contain(Tags.Env, "Overridden Environment");
             deserializedSpan.Tags.Should().Contain(Tags.Language, TracerConstants.Language);
             deserializedSpan.Tags.Should().Contain(Tags.RuntimeId, Tracer.RuntimeId);
             deserializedSpan.Tags.Should().Contain(Tags.Propagated.DecisionMaker, $"-{SamplingMechanism.Default.ToString()}");
+            deserializedSpan.Tags.Should().HaveCount(customTagCount + 4);
 
-            deserializedSpan.Metrics.Count.Should().Be(customTagCount + 3);
+            deserializedSpan.Metrics.Should().Contain(Metrics.SamplingPriority, 1);
```

Comment:
Interesting, do you know why this didn't show up before?

Reply:
Yes, it's because the test now calls `Span.Finished()` on the spans before trying to serialize the trace chunk, and the `_sampling_priority_v1` metric is not added to the local root span until it is closed.

I had to close the spans in these tests because only when all open spans are closed does the `TraceContext` all `Tracer.WriteSpans(ArraySegment<Span>)` which I am relying on to get access to the trace chunk, since serializing individual spans doesn't make sense anymore.

Reply:
Note: after switching to using `MockApi`, I had to add allow for even more tags that were not present before:
- `SamplingPriority = "_sampling_priority_v1"` (which you noticed above)
- `SamplingAgentDecision = "_dd.agent_psr"`
- `TracesKeepRate = "_dd.tracer_kr"`